core/src/test/java/com/linecorp/armeria/common/RequestContextTest.java

```diff
@@ -307,6 +304,66 @@ public void makeContextAwareCompletableFutureWithAsyncChaining() throws Exceptio
         }
     }
 
+    @Test
+    public void minimalCompletionStageUsingCompleteAsync() throws Exception {
+        final RequestContext context = createContext(false);
+        final CompletableFuture<Integer> originalFuture = new CompletableFuture<>();
+        final RequestContextAwareCompletableFuture<Integer> contextAwareFuture = (RequestContextAwareCompletableFuture) context.makeContextAware(originalFuture);
+        final CompletionStage<Integer> completionStage = contextAwareFuture.minimalCompletionStage();
+
+        contextAwareFuture.completeAsync(() -> { throw new RuntimeException(); });
+        contextAwareFuture.complete(1);
+
+        assertThat(contextAwareFuture.join()).isEqualTo(1);
+        assertThat(contextAwareFuture.getNow(null)).isEqualTo(1);
+        assertThat(contextAwareFuture.get()).isEqualTo(1);
+        completionStage.toCompletableFuture().get();
+    }
+
+    @Test
+    public void minimalCompletionStageUsingWhenComplete() throws Exception {
+        final RequestContext context = createContext(false);
+        final CompletableFuture<Integer> originalFuture = new CompletableFuture<>();
+        final RequestContextAwareCompletableFuture<Integer> contextAwareFuture = (RequestContextAwareCompletableFuture) context.makeContextAware(originalFuture);
+        final CompletionStage<Integer> completionStage = contextAwareFuture.minimalCompletionStage();
+
+        AtomicInteger atomicInteger = new AtomicInteger(0);
+        AtomicReference<Throwable> atomicReference = new AtomicReference<>();
+        completionStage.whenComplete((v, e) -> {if (e != null) atomicReference.set(e); else atomicInteger.set(v);});
```

Comment:
Could you format properly by inserting newlines and indentations?

Reply:
Got it.